packages/shared-ini-file-loader/src/getHomeDir.ts

```diff
@@ -0,0 +1,17 @@
+import { homedir } from "os";
+import { sep } from "path";
+
+/**
+ * Get the HOME directory for the current runtime.
+ *
+ * @internal
+ */
+export const getHomeDir = (): string => {
+  const { HOME, USERPROFILE, HOMEPATH, HOMEDRIVE = `C:${sep}` } = process.env;
+
+  if (HOME) return HOME;
+  if (USERPROFILE) return USERPROFILE;
+  if (HOMEPATH) return `${HOMEDRIVE}${HOMEPATH}`;
```

Comment:
I hope this will work also for non-windows environments.

Reply:
This code is [existing code](https://github.com/aws/aws-sdk-js-v3/blob/63d7f8c539e0fd782fa5bf997bd6ac2730e4bfda/packages/shared-ini-file-loader/src/index.ts#L111-L124), and the PR just refactors it into a modular component for better code management.

Please create a [bug report](https://github.com/aws/aws-sdk-js-v3/issues/new?assignees=&labels=bug%2C+needs-triage&template=---bug-report.md&title=) if it's not working for specific non-windows environment and we'll triage it.